packages/pg-v5/lib/psql.js

```diff
@@ -49,9 +49,22 @@ function execPsqlWithFile (file, dbEnv) {
 function psqlInteractive (dbEnv, prompt) {
   const {spawn} = require('child_process')
   return new Promise((resolve, reject) => {
-    let psql = spawn('psql',
-      ['--set', `PROMPT1=${prompt}`, '--set', `PROMPT2=${prompt}`],
-      {env: dbEnv, stdio: 'inherit'})
+    let psql_args = ['--set', `PROMPT1=${prompt}`, '--set', `PROMPT2=${prompt}`, ]
+    let path = process.env.HEROKU_PG_HISTORY
+    if(path) {
+      const fs = require('fs')
+      if(fs.existsSync(path) && fs.statSync(path).isDirectory()) {
+        psql_args = psql_args.concat(['--set', `HISTFILE=${path}/${prompt.split(':')[0]}`])
+      }
+      else if (fs.existsSync(require('path').dirname(path))) {
+        psql_args = psql_args.concat(['--set', `HISTFILE=${path}`])
+      }
+      else {
+        console.warn(`HEROKU_PG_HISTORY is set but is not a valid path (${path})`)
```

Comment:
@mage2k Please correct me if I'm wrong, but if `HEROKU_PG_HISTORY` is not set correctly, it SHOULD NOT hinder the commands ability to work correctly, right?

Reply:
@elbandito Correct, and it does not hinder the command's ability to work.  console.warn does not stop program execution so if it is not set correctly that warning will print and the psql session will run with the system's default or user settings as would without the env variable set at all (i.e. as it does now).

Reply:
Ah, I see there's cli.warn from heroku-cli-util for printing warnings.  I switch it to that.